### PR TITLE
Added Destructuring example to “Destructuring assignment” doc

### LINKED ISSUE #4682: "Destructuring assignment" needs an example where the left-hand side has more variables than there are values on the right-hand side
const foo = ['one', 'two'];

const [red, yellow, orange, pink] = foo;
console.log(red); // "one"
console.log(yellow); // 
console.log(orange); // undefined
console.log(pink); // undefined

### PR DESCRIPTION
**Pull request by NegiAkash890**
This commit adds an example to the file.

FIxes #4682 

Now:

Screenshot from 2021-05-20 15-56-12

### CODE REVIEW

**Review by sideshowbarker — COMMENTED** [review]
You’ve added this in the **Assignment separate from declaration** section, but per https://github.com/mdn/content/issues/4682#issuecomment-843219785 it seems like you intended to add it in the **Basic variable assignment** section. So it should be moved up into that section.

File: files/en-us/web/javascript/reference/operators/destructuring_assignment/index.html [hunk 1]
```diff
@@ -91,6 +91,16 @@ <h4 id="Assignment_separate_from_declaration">Assignment separate from declarati
 console.log(b); // 2
 </pre>
 
+<p>If there are fewer items in the array than the variables specified during assignment, then destructuring automatically assigns default value(<code>undefined</code>) to the extra variables:</p>
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/operators/destructuring_assignment/index.html:94]:
```suggestion
<p>In an array destructuring from an array of length <var>N</var> specified on the right-hand side of the assignment, if the number of variables specified on the left-hand side of the assignment is greater than <var>N</var>, only the first <var>N</var> variables are assigned values. The values of the remaining variables will be undefined.</p>
```

File: files/en-us/web/javascript/reference/operators/destructuring_assignment/index.html [hunk 2]
```diff
@@ -91,6 +91,16 @@ <h4 id="Assignment_separate_from_declaration">Assignment separate from declarati
...
+<pre class="brush: js">const foo = ['one', 'two'];
+
+const [red, yellow, green,blue] = foo;
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/operators/destructuring_assignment/index.html:97]:
```suggestion
const [red, yellow, green, blue] = foo;
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

**Review by sideshowbarker — COMMENTED** [review]
The comment at https://github.com/mdn/content/issues/4682#issuecomment-831694980 also suggested an example of object destructuring, and at https://github.com/mdn/content/issues/4682#issuecomment-831838837, Chris suggested an idea for that example — but this PR doesn’t include an example which addresses those comments.

**Comment by NegiAkash890** [comment]
Hi @sideshowbarker thanks for the review !
For the second example :

```
function FullName({firstName,lastName}){
  console.log('Welcome ' + firstName + '  ' + lastName + ' to mdn docs!' )
}

//Calling Function Method1
FullName({firstName:"John",lastName:"Doe"})

or

//Calling function Method2
let person={
firstName:"John",
lastName:"Doe"
}
FullName(person)                      //This looks clean


```
Further extension to the above example : 
In this example the `firstName` and 'lastName' are destructured
```
function FullName({firstName,lastName},age){
  console.log('Welcome ' + firstName + '  ' + lastName + ' to mdn docs!' )
  console.log("You are " + age + "years old")
}

//Calling function Method2
FullName({firstName:"John",lastName:"Doe"},10) 

or

//Calling function Method2
let person={
firstName:"John",
lastName:"Doe"
}
FullName(person,10)                  //This looks clean

```
Please let me know your views on the above example after  that I will be making final changes and also in which section should these be placed so it fits best with the docs ??

Thanks

**Comment by sideshowbarker** [comment]
> Please let me know your views on the above example

Can you explain that the example is intended to illustrate?

The purpose of the array example you added in this PR was to show what happens when there are parameters in the left-hand side of the destructuring assignment which don’t map to anything in the right-hand side of the assignment.

So, as far as the second part of the request in https://github.com/mdn/content/issues/4682, wasn’t that  a request for showing what happens when you have parameters in the left-hand side of an assignment from an object that don’t map to anything in the object on the right-hand side of the assignment.

So the example suggested in https://github.com/mdn/content/issues/4682#issuecomment-831694980 was for something like this:

```js
const user = {
id: 42
}

const {id, is_verified} = user;

console.log(id); // 42
console.log(is_verified); // undefined
```

**Comment by NegiAkash890** [comment]
> > Please let me know your views on the above example
> 
> Can you explain that the example is intended to illustrate?
> 
> The purpose of the array example you added in this PR was to show what happens when there are parameters in the left-hand side of the destructuring assignment which don’t map to anything in the right-hand side of the assignment.
> 
> So, as far as the second part of the request in #4682, wasn’t that a request for showing what happens when you have parameters in the left-hand side of an assignment from an object that don’t map to anything in the object on the right-hand side of the assignment.
> 
> So the example suggested in [#4682 (comment)](https://github.com/mdn/content/issues/4682#issuecomment-831694980) was for something like this:
> 
> ```js
> const user = {
> id: 42
> }
> 
> const {id, is_verified} = user;
> 
> console.log(id); // 42
> console.log(is_verified); // undefined
> ```

Hi @sideshowbarker , thanks for the review : )  
I will also make the necessary changes that were mentioned in  .

The changes that I am proposing above were asked by Chris .
Please follow up the conversion  here.

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
I’m going to go ahead and merge this but keep #4682 open, so that it can be resolved by follow-up PR that adds an object destructuring example.

**Pull request merged**